presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java

```diff
@@ -376,8 +377,13 @@ public static void renameDirectory(HdfsEnvironment hdfsEnvironment, String schem
     public static Path createTemporaryPath(HdfsEnvironment hdfsEnvironment, Path targetPath)
     {
         // use a per-user temporary directory to avoid permission problems
-        // TODO: this should use Hadoop UserGroupInformation
-        String temporaryPrefix = "/tmp/presto-" + StandardSystemProperty.USER_NAME.value();
+        String temporaryPrefix;
+        try {
+            temporaryPrefix = "/tmp/presto-" + UserGroupInformation.getCurrentUser().getUserName();
```

Comment:
Answering my own question here since /tmp is executable by all. The issue we run into is that /tmp and /user are on different viewfs filesystems so we can't easily copy from /tmp to /user. Changing this to create the tmp/ dir in the user home could be a different patch of ours.